### PR TITLE
[CI] Optionally run firebase tests only on non-main, non-release branches

### PR DESCRIPTION
**Pull request by MarkVillacampa**
That way we avoid keeping orange all commits in main.

### CODE REVIEW

**Review by NachoSoto — APPROVED** [review]
Thanks!

File: .circleci/config.yml [hunk 1]
```diff
@@ -381,11 +372,11 @@ workflows:
           type: approval
           requires:
             - android-integration-test-build
-          <<: *release-branches-and-main
+          <<: *release-branches
       - run-firebase-tests:
           requires:
             - hold
-          <<: *release-branches-and-main
+          <<: *release-branches
```
Comment (tonidero) [inline .circleci/config.yml:379]: Hmm I think we should probably just remove this. Note that we are already running the integration tests on release branches as part of the `deploy` workflow. IMO, I think this should be running in the `main` branch only. Thoughts?

File: .circleci/config.yml [hunk 2]
```diff
@@ -381,11 +379,11 @@ workflows:
           type: approval
           requires:
             - android-integration-test-build
-          <<: *release-branches-and-main
+          <<: *except-main-and-release
       - run-firebase-tests:
           requires:
             - hold
-          <<: *release-branches-and-main
+          <<: *except-main-and-release
```
Comment (MarkVillacampa) [inline .circleci/config.yml:386]: With this change, firebase tests will run:
- Never in main branch
- Always on release branches
- At least once per PR (if we want to have the green check before merging)

**Review by tonidero — APPROVED** [review]
Nice!

**Pull request merged**